software/firmware/ui.py

```diff
@@ -43,9 +42,13 @@ def draw_menu(self):
         """This function should be called by your script's main loop in order to display and refresh the menu."""
         current = self.selected
         oled.fill(0)
-        oled.text(f"{self.items[current - 1]}", 2, 3, 1)
-        self._inverted_text(f"{self.items[current]}", 2, 13)
-        if current != len(self.items) - 2:
-            # don't show the title at the bottom of the menu
-            oled.text(f"{self.items[current + 1]}", 2, 23, 1)
+        line_height = CHAR_HEIGHT + 2
+
+        for line_number, item in enumerate(self.items):
+            y_position = ((line_number - current) * line_height) + 1
+            if line_number == (current + 1):
```

Comment:
[optional] It feels a little 'wrong' that this attempts to draw the _entire_ menu, even items that are not on the screen. Also it's a little weird that you can see the top pixels of the next item. I think that you should be able to solve both of these problems by adding an initial clause to this `if` block that ignores items based on their `y_position`.

Reply:
Done, now only full lines will be displayed, no clipped pixels